src/realm/util/encrypted_file_mapping.cpp

```diff
@@ -338,8 +413,10 @@ EncryptedFileMapping::EncryptedFileMapping(SharedFileInfo& file, size_t file_off
 
 EncryptedFileMapping::~EncryptedFileMapping()
 {
-    flush();
-    sync();
+    if (m_access & File::access_ReadWrite) {
```

Comment:
This technically works right now, but I would prefer `m_access == File::access_ReadWrite` because an incrementing enum value is not guaranteed to be bit exclusive.